src-electron/generator/helper-zcl.js

```diff
@@ -2456,6 +2456,160 @@ async function format_zcl_string_as_characters_for_generated_defaults(
+      let bitmap = await queryZcl.selectBitmapByName(
+        this.global.db,
+        packageIds,
+        dataType.name
+      )
+      return Math.pow(2, Math.ceil(Math.log2(bitmap.size))) * sizeMultiple
+    } else if (
+      dataType.discriminatorName.toLowerCase() == dbEnum.zclType.enum
+    ) {
+      let en = await queryZcl.selectEnumByName(
+        this.global.db,
+        dataType.name,
+        packageIds
+      )
+      return Math.pow(2, Math.ceil(Math.log2(en.size))) * sizeMultiple
+    } else if (
+      dataType.discriminatorName.toLowerCase() == dbEnum.zclType.number
+    ) {
+      let number = await queryZcl.selectNumberByName(
+        this.global.db,
+        packageIds,
+        dataType.name
+      )
+      return Math.pow(2, Math.ceil(Math.log2(number.size))) * sizeMultiple
+    } else if (
+      dataType.discriminatorName.toLowerCase() == dbEnum.zclType.struct
+    ) {
+      return 0
+    } else if (
+      dataType.discriminatorName.toLowerCase() == dbEnum.zclType.string
+    ) {
+      return 0
+    } else {
+      return 0
+    }
+  } else {
+    return 0
+  }
+}
+
+/**
+ * If helper that checks if a type is signed or not.
+ * All zcl data types except numbers are considered unsigned
+ * example:
+ * {{#if_is_underlying_zcl_type_signed type}}
+ * type is signed
+ * {{else}}
+ * type is not signed
+ * {{/if_is_underlying_zcl_type_signed}}
+ *
+ * @param {*} type
+ * @param {*} options
+ * @returns Promise of content.
+ */
+async function if_is_underlying_zcl_type_signed(type, options) {
+  const packageIds = await templateUtil.ensureZclPackageIds(this)
+  let dataType = await queryZcl.selectDataTypeByName(
+    this.global.db,
+    type,
+    packageIds
+  )
+
+  if (
+    dataType &&
+    dataType.discriminatorName.toLowerCase() == dbEnum.zclType.number
+  ) {
+    let number = await queryZcl.selectNumberByName(
+      this.global.db,
+      packageIds,
+      dataType.name
+    )
+    if (number.isSigned) {
+      return options.fn(this)
+    } else {
+      return options.inverse(this)
+    }
+  } else {
+    return options.inverse(this)
+  }
+}
+
+/**
+ * If helper that checks if a type is either float or double
+ *
+ * example:
+ * {{#if_is_float_or_double_data_type type}}
+ * type is float or double
+ * {{else}}
+ * type is not float or double
+ * {{/if_is_float_or_double_data_type}}
+ *
+ * @param {*} type
+ * @returns Promise of content.
+ */
+async function if_is_float_or_double_data_type(type, options) {
+  let packageIds = await templateUtil.ensureZclPackageIds(this)
+  let number = ''
+  if (type && typeof type === 'string') {
+    number = await queryZcl.selectNumberByName(
+      this.global.db,
+      packageIds,
+      type.toLowerCase()
+    )
+  } else {
+    number = await queryZcl.selectNumberById(this.global.db, type)
+  }
+
+  if (
+    number &&
+    (number.name.includes('float') ||
+      number.name.includes('double') ||
+      number.name.includes('single'))
```

Comment:
This can't end up being a struct/enum/bitmap type with a name that happens to include those strings?